TheThingsNetwork/lorawan-stack
# Support batch unclaiming of end devices

#### Summary



Closes https://github.com/TheThingsIndustries/lorawan-stack/issues/3666

#### Changes



- Change `GetClaimStatus` and `Unclaim` to fetch EUIs from IS instead of using the values in the request.
- Add RPC and implementation for batching.
- Tests
- Update `pkg/identityserver/mock`

#### Testing



1. Connect TTS to [The Things Join Server](https://www.thethingsindustries.com/docs/reference/interop-repository/#interoperability-with-the-things-join-server)
2. Register multiple devices with a Join EUI that can be  claimed on TTJS.
3. Setup the CLI and run the `batch-delete` command. 
```
$ ttn-lw-cli devices batch-delete <app-id> <dev-id-1> <dev-id-2> ...
```
4. Check that the devices were successfully unclaimed.

See the first comment for testing results.

##### Regressions



The steps are the same for testing regressions. Test the following for regressions:
1. Unclaiming (not batch)
  - Same steps as above, except use the `delete` command for end devices.
  - This should be covered by existing Cypress claiming tests.

2. GetClaimInfo (We don't support this in the CLI so this has to be via Postman)
  -  Same setup steps, before deleting a claimed device query it's status.

#### Notes for Reviewers



@johanstokking: Is there a way for us to have a TTJS dev environment to test?

#### Checklist
